decidim-elections/lib/decidim/elections/bulletin_board_client.rb

```diff
@@ -6,29 +6,51 @@ class BulletinBoardClient
       def initialize(params)
         @server = params[:server].presence
         @api_key = params[:api_key].presence
-        @identification_private_key = params[:identification_private_key]&.strip.presence
+        @scheme = params[:scheme].presence
+        @authority_name = params[:authority_name].presence
+        @number_of_trustees = 0
+        @number_of_trustees = params[:number_of_trustees] if params[:number_of_trustees].present?
+        @identification_private_key = params[:identification_private_key].presence
+        @private_key = identification_private_key_content if identification_private_key
+      end
+
+      attr_reader :server, :scheme, :api_key, :number_of_trustees, :authority_name
+
+      def quorum
```

Comment:
All settings related to voting schemes should be checked on the BB side, or by an specific class for the voting scheme in Decidim. As we are going to refactor this client and move it to the BB repo, we can change this at that moment, but it's important to have it in mind.